### PR TITLE
Fix spacy CI

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/13341?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13341/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13341
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Fix spacy CI
Spacy >=3.8.2 requires numpy >= 2.0, which doesn't support python < 3.9 now. We need to set python version to 3.9 for it

For fastai (it has spacy dependency) cross tests, migrating it to python 3.9 has other issues (https://github.com/mlflow/mlflow/actions/runs/11227490874/job/31209816918?pr=13341#step:8:173 ) I will keep running py3.8 for it. but restrict Spacy version to < 3.8.2

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -255,6 +255,7 @@ fastai:
     minimum: "2.4.1"
     maximum: "2.7.17"
     requirements:
+      "> 0.0.0": ["spacy<3.8.2"]
```
[the remark this answers is not in the document]
Comment (serena-ruan) [reply mlflow/ml-package-versions.yml:258]: Can we file an issue to fastai to track this? Seems like spacy is introduced by fastai, then they should fix the issue for python 3.9

[the remark this answers is not in the document]
Comment (WeichenXu123) [reply mlflow/ml-package-versions.yml:258]: sure

File: mlflow/ml-package-versions.yml [hunk 2]
```diff
@@ -255,6 +255,9 @@ fastai:
     minimum: "2.4.1"
     maximum: "2.7.17"
     requirements:
+      # TODO: fastai depends on spacy, but spacy>=3.8.2 requires python>=3.9
+      #  Once migrating fastai tests to python 3.9, we should remove spacy<3.8.2
```
Comment (serena-ruan) [inline mlflow/ml-package-versions.yml:259]:
```suggestion
      # Once migrating fastai tests to python 3.9, we should remove spacy<3.8.2
```
Let's also link the filed issue here :)

**Review by serena-ruan — APPROVED** [review]
LGTM!

File: mlflow/ml-package-versions.yml [hunk 3]
```diff
@@ -255,6 +255,12 @@ fastai:
...
+      #  To migrate fastai tests to python 3.9, we need to address error
+      #  when installing old version scipy: "No module named 'distutils.msvccompiler'"
+      #  similar issue link: https://github.com/numpy/numpy/issues/27405
```
Comment (WeichenXu123) [inline mlflow/ml-package-versions.yml:262]: this needs to pin setuptools to old version, which requires to modify CI python setup code, can be done in follow-up PR.

**Pull request merged**